src/battleship/game/grid.go

```diff
@@ -0,0 +1,101 @@
+package game
+
+import (
+	"battleship/scoreboard"
+)
+
+// Grid is the board on which the battle ships are positionned
+type Grid struct {
+	Size  int
+	Ships []Ship
+	Cells []Cell
+}
+
+// NewGrid create a square grid of the given size
+func NewGrid(gridSize int) (Grid, error) {
+	cells := []Cell{}
+
+	for row := 0; row < gridSize; row++ {
+		for column := 0; column < gridSize; column++ {
+			cells = append(cells, Cell{row, column})
+		}
+	}
+
+	return Grid{gridSize, []Ship{}, cells}, nil
+}
+
+// AddShip add a ship to the grid
+func AddShip(grid Grid, ship Ship) Grid {
+	grid.Ships = append(grid.Ships, ship)
+	return grid
+}
+
+// GetScoreBoard returns the score board of a ship positioned on a grid of the given size
+func GetScoreBoard(grid Grid, ship Ship) (*scoreboard.ScoreBoard, error) {
```

Comment:
error is not used as well

Reply:
why this method is in that file ?

Reply:
I have some difficulties to organize the files because I find it more meaningfull to call `grid.GetScoreBoard` instead of `scoreboard.GetScoreBoard`. I will move the method to its file as I do not have better counter argument ^^